htlcswitch/link.go

```diff
@@ -33,14 +33,93 @@ const (
 )
 
 var (
-	// ErrInternalLinkFailure is a generic error returned to the remote
-	// party so as to obfuscate the true failure.
-	ErrInternalLinkFailure = errors.New("internal link failure")
-
 	// ErrLinkShuttingDown signals that the link is shutting down.
 	ErrLinkShuttingDown = errors.New("link shutting down")
 )
 
+// errorCode encodes the possible types of errors that will make us fail the
```

Comment:
Maybe justified in having it's own file?

Reply:
Done :)